packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -95,110 +47,99 @@ int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
         break;
     }
 
-    // contruct a cloud of points (nodes of a structured grid)
-    auto cloud = make_stuctured_cloud( Lx, Ly, Lz, nx, ny, nz );
-    int n = cloud.size();
-
-    Kokkos::View<DataTransferKit::Box *, DeviceType> bounding_boxes(
-        "bounding_boxes", n );
-    auto bounding_boxes_host = Kokkos::create_mirror_view( bounding_boxes );
-    // build bounding volume hierarchy
-    for ( int i = 0; i < n; ++i )
+    Kokkos::View<DataTransferKit::Point *, DeviceType> random_points(
+        "random_points" );
     {
-        auto const &point = cloud[i];
-        double x = std::get<0>( point );
-        double y = std::get<1>( point );
-        double z = std::get<2>( point );
-        bounding_boxes_host[i] = {{{x, y, z}}, {{x, y, z}}};
+        auto n = std::max( n_values, n_queries );
+        Kokkos::resize( random_points, n );
+
+        auto random_points_host = Kokkos::create_mirror_view( random_points );
+
+        // edge length chosen such that object density will remain constant as
+        // problem size is changed
+        auto const a = std::cbrt( n_values );
+        std::uniform_real_distribution<double> distribution( -a, +a );
+        std::default_random_engine generator;
+        auto random = [&distribution, &generator]() {
+            return distribution( generator );
+        };
+        for ( int i = 0; i < n; ++i )
+            random_points_host( i ) = {random(), random(), random()};
+        Kokkos::deep_copy( random_points, random_points_host );
     }
-    Kokkos::deep_copy( bounding_boxes, bounding_boxes_host );
 
+    Kokkos::View<DataTransferKit::Box *, DeviceType> bounding_boxes(
+        "bounding_boxes", n_values );
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, n_values ),
+                          KOKKOS_LAMBDA( int i ) {
+                              double const x = random_points( i )[0];
+                              double const y = random_points( i )[1];
+                              double const z = random_points( i )[2];
+                              bounding_boxes( i ) = {
+                                  {{x - 1., y - 1., z - 1.}},
+                                  {{x + 1., y + 1., z + 1.}}};
+                          } );
+    Kokkos::fence();
+
+    std::ostream &os = std::cout;
+
+    auto start = std::chrono::high_resolution_clock::now();
     DataTransferKit::BVH<DeviceType> bvh( bounding_boxes );
```

Comment:
Does Kokkos automatically launch a little kernel to "warm up" the device? I know there is always some type of overhead with the first kernel launch in a program. This likely does not even come close to accounting for the scaling of the construction using CUDA but would be interesting to know. If I remember I think they may call a small no-op kernel upon calling initialize.

Reply:
See [`Kokkos::Impl::CudaInternal::initialize()`](https://github.com/kokkos/kokkos/blob/master/core/src/Cuda/Kokkos_Cuda_Impl.cpp#L370-L564).